### DISCUSSION TITLE
服务长时间不返回客户端调用直接异常结束

### DISCUSSION THREAD

**Discussion by jjqcat** [alibaba/yalantinglibs#618] [Q&A] [answered]
环境：windows10 X64, VS2022
服务端算法比较复杂，运算可能有时超过20秒，调用服务端函数，服务端还未结束并返回，客户端直接结束了。
服务端注册服务 async_simple::coro::Lazy<类型> coro_Work(参数);

**Comment by qicosmos** [comment, marked as answer]
客户端默认超时时间是5秒，你可以把调用的超时时间改成20秒或更长。

参考这个代码：
https://github.com/alibaba/yalantinglibs/blob/main/src/coro_rpc/tests/test_coro_rpc_client.cpp#L134

```c++
co_await client->call_for< coro_Work >(30s);
```

**Comment by jjqcat** [comment]
谢谢，这个问题我测试下。还有一个疑问，async_simple::coro::Lazy<类型> coro_Work(输入参数); 输入参数是一个结构体，这个结构体大小最大可以是多少字节？ 之前碰到stack error,把结构体定义改小才可以

**Comment by poor-circle** [comment]
我理解这和你的编译器分配的栈大小有关系吧

**Reply by jjqcat** [reply]
这个结构体200KB左右，按理说不至于超出系统限制，可能跟之前调用5s这个有关，我重新测试下

**Reply by poor-circle** [reply]
如果遇到栈溢出问题可以反馈给我们，理论上来说，coro_rpc会把你传入的参数的引用打包成一个tuple，然后用struct_pack序列化这个tuple保存到一个std::string里面，所以应该是没有什么限制的。
